src/net.cpp

```diff
@@ -1967,7 +1972,18 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             CAddrInfo addr = addrman.SelectTriedCollision();
 
             // SelectTriedCollision returns an invalid address if it is empty.
-            if (!fFeeler || !addr.IsValid()) {
+            bool test_before_evict = fFeeler && addr.IsValid();
+
+            // If test-before-evict logic would have us connect to a peer that
+            // we're already connected to, just mark that address as Good(). We
+            // won't be able to initiate the connection anyway, so this avoids
+            // inadvertently evicting a currently-connected peer.
+            if (test_before_evict && AlreadyConnectedToPeer(addr)) {
+                MarkAddressGood(addr);
```

Comment:
No need to call this CConMan forwarding function. Just call `addrman.Good()` directly.

Reply:
I took this change as part of your patch for rewriting the loop.